src/overlays/actors/ovl_En_Syateki_Man/z_en_syateki_man.c

```diff
@@ -15,24 +21,43 @@ void EnSyatekiMan_Destroy(Actor* thisx, PlayState* play);
 void EnSyatekiMan_Update(Actor* thisx, PlayState* play);
 void EnSyatekiMan_Draw(Actor* thisx, PlayState* play);
 
-void func_809C6810(EnSyatekiMan* this, PlayState* play);
-void func_809C6848(EnSyatekiMan* this, PlayState* play);
-void func_809C6E30(EnSyatekiMan* this, PlayState* play);
-void func_809C72D8(EnSyatekiMan* this, PlayState* play);
-void func_809C7990(EnSyatekiMan* this, PlayState* play);
-void func_809C7A90(EnSyatekiMan* this, PlayState* play);
-void func_809C7C14(EnSyatekiMan* this, PlayState* play);
-void func_809C7D14(EnSyatekiMan* this, PlayState* play);
-void func_809C7EB4(EnSyatekiMan* this, PlayState* play);
-void func_809C7FFC(EnSyatekiMan* this, PlayState* play);
-void func_809C80C0(EnSyatekiMan* this, PlayState* play);
-void func_809C81D0(EnSyatekiMan* this, PlayState* play);
-void func_809C8488(EnSyatekiMan* this, PlayState* play);
-void func_809C8610(EnSyatekiMan* this, PlayState* play);
-void func_809C8710(EnSyatekiMan* this, PlayState* play);
-void func_809C8808(EnSyatekiMan* this, PlayState* play);
-void func_809C898C(EnSyatekiMan* this, PlayState* play);
-void func_809C8BF0(EnSyatekiMan* this, PlayState* play);
+void EnSyatekiMan_SetupIdle(EnSyatekiMan* this, PlayState* play);
+void EnSyatekiMan_Swamp_Idle(EnSyatekiMan* this, PlayState* play);
+void EnSyatekiMan_Swamp_Talk(EnSyatekiMan* this, PlayState* play);
+void EnSyatekiMan_Town_Idle(EnSyatekiMan* this, PlayState* play);
+void EnSyatekiMan_Town_Talk(EnSyatekiMan* this, PlayState* play);
+void EnSyatekiMan_Swamp_SetupGiveReward(EnSyatekiMan* this, PlayState* play);
+void EnSyatekiMan_Swamp_GiveReward(EnSyatekiMan* this, PlayState* play);
+void EnSyatekiMan_Town_SetupGiveReward(EnSyatekiMan* this, PlayState* play);
+void EnSyatekiMan_Town_GiveReward(EnSyatekiMan* this, PlayState* play);
+void EnSyatekiMan_Swamp_MovePlayerAndExplainRules(EnSyatekiMan* this, PlayState* play);
+void EnSyatekiMan_Swamp_StartGame(EnSyatekiMan* this, PlayState* play);
+void EnSyatekiMan_Swamp_RunGame(EnSyatekiMan* this, PlayState* play);
+void EnSyatekiMan_Swamp_EndGame(EnSyatekiMan* this, PlayState* play);
+void EnSyatekiMan_Swamp_AddBonusPoints(EnSyatekiMan* this, PlayState* play);
+void EnSyatekiMan_Town_MovePlayerAndSayHighScore(EnSyatekiMan* this, PlayState* play);
+void EnSyatekiMan_Town_StartGame(EnSyatekiMan* this, PlayState* play);
+void EnSyatekiMan_Town_RunGame(EnSyatekiMan* this, PlayState* play);
+void EnSyatekiMan_Town_EndGame(EnSyatekiMan* this, PlayState* play);
+
+#define TALK_FLAG_NONE 0
+#define TALK_FLAG_TOWN_HAS_SPOKEN_WITH_HUMAN (1 << 0)
+#define TALK_FLAG_TOWN_HAS_SPOKEN_WITH_DEKU (1 << 1)
+#define TALK_FLAG_TOWN_HAS_SPOKEN_WITH_GORON (1 << 2)
+#define TALK_FLAG_TOWN_HAS_SPOKEN_WITH_ZORA (1 << 3)
+#define TALK_FLAG_TOWN_HAS_EXPLAINED_THE_RULES (1 << 4)
+#define TALK_FLAG_SWAMP_HAS_SPOKEN_WITH_HUMAN (1 << 0)
+#define TALK_FLAG_SWAMP_HAS_EXPLAINED_THE_RULES (1 << 1)
```

Comment:
Since the Shooting gallery handles all instances, would it make sense to only have 1 flag for `HAS_SPOKEN_WITH_HUMAN`? I'm not too familiar if there is any major differences or if its best to keep two separate flags.

Reply:
My logic for splitting them apart is that the town shooting gallery man uses bitwise operations with these flags. For example:
```c
    if (!(this->talkFlags & TALK_FLAG_TOWN_HAS_SPOKEN_WITH_HUMAN)) {
        this->talkFlags |= TALK_FLAG_TOWN_HAS_SPOKEN_WITH_HUMAN;
```
Whereas the swamp shooting gallery man just does raw value checks. For example:
```c
    if (this->talkFlags == TALK_FLAG_NONE) {
        this->talkFlags = TALK_FLAG_SWAMP_HAS_SPOKEN_WITH_HUMAN;
```
I thought that splitting them up like this would make sense given the different operations; for the swamp shooting gallery man, it's almost like these aren't "flags" at all. However, I totally see your point in that the two values are exactly the same and represent basically the same thing. I'd be willing to switch over to a single `#define` for `HAS_SPOKEN_WITH_HUMAN`, but I'm curious what you (and others) think now that I've explained my thoughts.

Reply:
I think splitting them completely makes the most sense, though don't really have too strong an opinion.